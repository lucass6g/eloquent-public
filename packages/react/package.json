{
  "name": "@eloquent/react",
  "version": "1.0.0",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup",
    "clean": "rm -rf dist && rm -rf node_modules"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.1.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "react": "^18.3.1",
    "typescript": "^5.4.5"
  }
}
