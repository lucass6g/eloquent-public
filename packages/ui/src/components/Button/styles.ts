import { cva } from 'class-variance-authority';

export type ClassVariantsProps = Readonly<{
	variant: {
		primary?: string[]
		secondary?: string[]
		link?: string[]
		icon?: string[]
	}
	size: {
		small?: string[]
		medium?: string[]
		large?: string[]
		icon?: string[]
	}
	inverse: {
		true?: string[]
		false?: string[]
	}
	fullWidth: {
		true?: string[]
		false?: string[]
	}
}>

export const buttonVariants = cva<ClassVariantsProps>(
	[
		'font-sans text-xs font-semibold',
		'outline-offset-4',
		'box-border border-2',
		'transition-all',
		'disabled:opacity-50 disabled:border-none disabled:cursor-not-allowed ',
	],
	{
		variants: {
			variant: {
				primary: [
					'text-neutral-100',
					'border-transparent',
					'bg-positive-100',
					'hover:bg-neutral-200',
					'active:bg-base-light-01 active:text-base-dark-01',
				],
				secondary: [
					'bg-neutral-dark-100',
					'text-neutral-dark-1000',
					'border-base-light-02',
					'hover:bg-base-light-02',
					'active:border-base-light-02 bg-base-light-01',
				],
				link: ['text-primary underline-offset-4 hover:underline'],
				icon: [
					'bg-transparent border-transparent font-normal border-none',
					'flex  flex-row  gap-stack-xxs',
				],
			},
			size: {
				small: ['px-inline-xxxs h-8 leading-tiny text-xxs'],
				medium: ['px-inline-xxs h-12 leading-default'],
				large: ['px-inline-xs h-14 leading-distant text-sm'],
				icon: [],
			},
			inverse: {
				true: [''],
			},
			fullWidth: {
				true: ['w-full'],
			},
		},
		defaultVariants: {
			variant: 'primary',
			size: 'medium',
		},
		compoundVariants: [
			{
				variant: 'primary',
				inverse: true,
				class: [
					'bg-neutral-100',
					'text-neutral-1000',
					'hover:bg-base-light-02',
					'active:bg-base-light-03 active:text-base-dark-03',
				],
			},
			{
				variant: 'secondary',
				inverse: true,
				class: [
					'bg-transparent',
					'border-neutral-100',
					'text-neutral-100',
					'hover:bg-base-dark-03',
					'active:bg-base-dark-01 active:text-base-light-01',
					'disabled:bg-neutral-1000 disabled:opacity-50 disabled:border-base-light-02',
				],
			},
		],
	},
)
