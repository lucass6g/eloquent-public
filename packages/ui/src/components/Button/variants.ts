import {tv} from "tailwind-variants"

export const baseButton = tv({
    base: [
        "inline-flex items-center justify-center whitespace-nowrap",
        "border-lg rounded-lg transition-colors",
        "focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1",
        "disabled:pointer-events-none",
        "text-sm font-medium antialiased",
        "leading-4"
    ]
})

export const buttonVariants = tv({
    extend: baseButton,
    variants: {
        variant: {
            default:[
                "bg-primary text-primary-foreground shadow",
                "hover:bg-green-dark-400",
                "disabled:bg-neutral-300 disabled:text-neutral-500 disabled:font-medium",
            ],
            secondary: [
                "shadow-sm",
                "border-foreground",
                "text-accent-foreground hover:bg-accent hover:text-accent-foreground",
            ],
            destructive:
                "bg-destructive text-destructive-foreground shadow-sm hover:opacity-90",
            link: "text-primary underline-offset-4 hover:underline",
            icon: "p-0",
            loading: "",
        },
        size: {
            default: "py-2 px-4 min-w-[152px] min-h-[24px] border-lg",
            sm: "h-8 px-3 text-xs",
            icon: "h-9 w-9 first-child:stroke-primary-foreground",
        },
    },
    defaultVariants: {
        variant: "default",
        size: "default",
    },
});