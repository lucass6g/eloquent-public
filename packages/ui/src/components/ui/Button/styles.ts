import { cva } from 'class-variance-authority';

export type ClassVariantsProps = Readonly<{
	variant: {
		primary?: string[]
		secondary?: string[]
		link?: string[]
		icon?: string[]
	}
	size: {
		small?: string[]
		medium?: string[]
		large?: string[]
		icon?: string[]
	}
	inverse: {
		true?: string[]
		false?: string[]
	}
	fullWidth: {
		true?: string[]
		false?: string[]
	}
}>

export const buttonVariants = cva<ClassVariantsProps>(
	[
		'font-sans text-xs font-semibold',
		'outline-offset-4',
		'box-border border-2',
		'transition-all',
		'disabled:opacity-50 disabled:border-none disabled:cursor-not-allowed ',
	],
	{
		variants: {
			variant: {
				primary: [
					'text-base-light-pure',
					'border-transparent',
					'bg-base-dark-pure',
					'hover:bg-base-dark-02',
					'active:bg-base-light-01 active:text-base-dark-01',
				],
				secondary: [
					'bg-base-dark-pure',
					'text-base-dark-pure',
					'border-base-light-02',
					'hover:bg-base-light-02',
					'active:border-base-light-02 bg-base-light-01',
				],
				link: ['text-primary underline-offset-4 hover:underline'],
				icon: [
					'bg-transparent border-transparent font-normal border-none',
					'flex  flex-row  gap-stack-xxs',
				],
			},
			size: {
				small: ['px-inline-xxxs h-8 leading-tiny text-xxs'],
				medium: ['px-inline-xxs h-12 leading-default'],
				large: ['px-inline-xs h-14 leading-distant text-sm'],
				icon: [],
			},
			inverse: {
				true: [''],
			},
			fullWidth: {
				true: ['w-full'],
			},
		},
		defaultVariants: {
			variant: 'primary',
			size: 'medium',
		},
		compoundVariants: [
			{
				variant: 'primary',
				inverse: true,
				class: [
					'bg-base-light-pure',
					'text-base-dark-pure',
					'hover:bg-base-light-02',
					'active:bg-base-light-03 active:text-base-dark-03',
				],
			},
			{
				variant: 'secondary',
				inverse: true,
				class: [
					'bg-transparent',
					'border-base-light-pure',
					'text-base-light-pure',
					'hover:bg-base-dark-03',
					'active:bg-base-dark-01 active:text-base-light-01',
					'disabled:bg-base-dark-pure disabled:opacity-50 disabled:border-base-light-02',
				],
			},
		],
	},
)
