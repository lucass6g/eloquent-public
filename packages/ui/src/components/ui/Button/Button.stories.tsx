import { Button }         from "./index";
import { Meta, StoryObj }          from "@storybook/react";

const meta: Meta<typeof Button> = {
  
  title: "Button",
  component: Button,
  parameters: {
    backgrounds: {
      default: "dark",
    },
  },
  args: {
    children: "Buttons",
    fullWidth: false,
    disabled: false,
  },
  argTypes: {
    disabled: {
      type:  "boolean"
    },
    size: {
      options: ['small', 'medium', 'large'],
      control: { type: "inline-radio" },
    },
    fullWidth: {
      control: { type: "boolean" },
    }
  }
};

export default meta;

type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  parameters: {
    backgrounds: {
      default: "light",
    },
  },
  args: {
    variant: "primary",
    inverse: false,
  },
  argTypes: {
    variant: {
      options: ['primary'],
      control: { type: "inline-radio" },
    }
  }
};

export const PrimaryInverse: Story = {
  args: {
    ...Primary.args,
    inverse: true,
  },
  argTypes: {
    variant: {
      options: ['primary'],
      control: { type: "inline-radio" },
    }
  }
};

export const Secondary: Story = {
  parameters: {
    backgrounds: {
      default: "light",
    },
  },
  args: {
    variant: "secondary",
  },
  argTypes: {
    variant: {
      options: ['secondary'],
      control: { type: "inline-radio" },
    }
  }
};

export const SecondaryInverse: Story = {
  parameters: {
    backgrounds: {
      default: "dark",
    },
  },
  args: {
    ...Secondary.args,
    inverse: true,
  },
  argTypes: {
    variant: {
      options: ['secondary'],
      control: { type: "inline-radio" },
    }
  }
};