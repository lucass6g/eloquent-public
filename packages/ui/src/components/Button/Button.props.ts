import { VariantProps } from "tailwind-variants";
import { buttonVariants } from "./variants";
import { ValueOf } from "src/types/select-key";

export interface ButtonHTMLAttributes extends React.ButtonHTMLAttributes<HTMLButtonElement>{}
export interface ButtonVariantsProps extends VariantProps<typeof buttonVariants>{}
export interface ButtonEloquentProps  {
    asChild?: boolean
}

export interface  ButtonProps extends ButtonEloquentProps, ButtonHTMLAttributes, ButtonVariantsProps {}

export type ButtonVariant = Required<ValueOf<ButtonVariantsProps, "variant">>;
export type ButtonSize = Required<ValueOf<ButtonVariantsProps, "size">>