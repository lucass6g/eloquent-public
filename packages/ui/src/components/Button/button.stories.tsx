import { Button } from ".";
import type { Meta, StoryObj } from "@storybook/react";
import { LayoutGrid, X } from "lucide-react";

import { ButtonProps, ButtonSize, ButtonVariant } from "./Button.props";

const meta: Meta<ButtonProps> = {
  title: "Button",
  component: Button,
  args: {
    children: "Clique aqui",
    disabled: false,
    size: "default",
    loading: false
  },
  argTypes: {
    onClick: {
      action: "onClick",
      table: {
        disable: true,
      },
    },
    onFocus: {
      action: "onFocus",
      table: {
        disable: true,
      },
    },
    onMouseOver: {
      action: "onMouseOver",
      table: {
        disable: true,
      },
    },
    type: {
      control: { type: "inline-radio" },
      options: ["button", "submit", "reset"],
      description: "Tipo do botão",
      table: {
        type: { summary: "string" },
        defaultValue: { summary: "button" },
      },
    },
    variant: {
      control: { type: "select" },
      options: [
        "default",
        "destructive",
        "link",
        "secondary",
        "badge",
        "none"
      ] satisfies ButtonVariant[],
      description: "Variante do botão",
      table: {
        type: { summary: "string" },
        defaultValue: { summary: "default" },
      },
    },
    size: {
      control: { type: "select" },
      options: ["default", "icon", "sm", "none"] satisfies ButtonSize[],
      description: "Tamanho do botão",
      table: {
        type: { summary: "string" },
        defaultValue: { summary: "default" },
      },
    },
    disabled: {
      description: "Se o botão está desabilitado",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: "false" },
      },
    },
    loading: {
      description: "Se o botão está em estado de `carregando...`",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: "false" },
      },
    },
    labelLoading: {
      description: "Texto que aparece quando o botão está em estado de carregando",
      table: {
        type: { summary: "string" },
        defaultValue: { summary: "Carregando..." },
      },
    }
  },
  tags: ["experimental"]
};

export default meta;

type Story = StoryObj<ButtonProps>;

export const Primary: Story = {
  args: {
    variant: "default",
  },
};

export const Secondary: Story = {
  args: {
    variant: "secondary",
  },
};

export const Link: Story = {
  args: {
    variant: "link",
  },
};

export const Destructive: Story = {
  args: {
    variant: "destructive",
  },
};

export const NoStyle: Story = {
  args: {
    variant: "none",
    size: "none",
  },
};

export const WithIcon: Story = {
  render(props) {
    return (
        <Button {...props}>
          <LayoutGrid className="w-4 h-4 mr-2"/>
          Clique aqui
        </Button>
    );
  }
};

export const Loading: Story = {
  render: (props) => {
    return (
      <Button {...props}>
        Clique aqui
      </Button>
    );
  },
  args: {
    loading: true
  }
}

export const Badge: Story = {
  render: (props) => {
    return (
          <Button {...props} variant="badge">
            Clique aqui
            <X className="ml-2 w-4 h-4" />
          </Button>
    );
  },
  args: {
    variant: "badge"
  }
}

export const Icon: Story = {
  render: ({children, ...props}) => {
    return(
      <Button {...props} size="icon" variant="icon">
        <LayoutGrid className="h-4 w-4" />
      </Button>
    )
  }
};
