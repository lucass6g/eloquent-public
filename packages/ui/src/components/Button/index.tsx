import * as React from "react";
import {Slot} from "@radix-ui/react-slot";
import {cn} from "@eloquent/styles";
import { ButtonProps } from "./Button.props";
import { buttonVariants } from "./variants";
import { Loader2 } from "lucide-react";

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
    function EloquentButton({
        className, 
        variant = "default", 
        disabled = false, 
        loading = false, 
        size = "default", 
        asChild = false, 
        children,
        labelLoading = "Carregando...",
        ...props
    }, 
        ref)  {
        const Comp = asChild ? Slot : "button";

        const LoadingContent = (
            <div className="flex items-center">
                <Loader2 className={cn("animate-spin h-4 w-4 mr-2", size === "icon" && "m-auto")}/>
                {size !== "icon" && labelLoading}
            </div>
        )
          
        return (
            <Comp
                disabled={disabled || loading}
                className={cn(variant === "none" ? buttonVariants({variant, className}): 
                buttonVariants({variant, size, className}))}
                ref={ref}
                {...props}
        >
            {loading ? LoadingContent : children}
        </Comp>
    );
    }
);
