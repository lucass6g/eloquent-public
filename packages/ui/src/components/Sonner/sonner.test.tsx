import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { it, describe, expect } from "vitest"
import { toast } from 'sonner';
import { SonnerToaster } from '.';
import { Button } from '../Button';
import { vi } from 'vitest';

describe('SonnerToaster', () => {
    beforeAll(() => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: vi.fn().mockImplementation((query) => ({
                matches: false,
                media: query,
                onchange: null,
                addEventListener: vi.fn(),
                removeEventListener: vi.fn(),
                dispatchEvent: vi.fn(),
            })),
        });
    });
    it('displays the toast message when the button is clicked', async () => {
        const toastMessageMock = vi.spyOn(toast, 'message');
        const component = render(
            <div>
                <SonnerToaster />
                <button onClick={() => toast.message('Elo o cart達o do brasileiro', {
                    description: 'Elo o cart達o do brasileiro',
                    action: <Button onClick={() => console.log(':)')}>Desfazer</Button>,
                })}>
                    Abrir
                </button>
            </div>
        );
        const button = screen.getByText('Abrir');
        fireEvent.click(button);
        await waitFor(() => {
            expect(toastMessageMock).toHaveBeenCalledWith('Elo o cart達o do brasileiro', {
                description: 'Elo o cart達o do brasileiro',
                action: expect.anything(),
            });
        });
        expect(component.baseElement).toMatchSnapshot();
        toastMessageMock.mockRestore();
    });
});