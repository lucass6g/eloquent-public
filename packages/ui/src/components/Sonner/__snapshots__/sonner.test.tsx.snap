// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SonnerToaster > displays the toast message when the button is clicked 1`] = `
<body>
  <div>
    <div>
      <section
        aria-label="Notifications alt+T"
        tabindex="-1"
      >
        <ol
          class="toaster group"
          data-sonner-toaster="true"
          data-theme="light"
          data-x-position="right"
          data-y-position="bottom"
          dir=""
          style="--front-toast-height: 0px; --offset: 32px; --width: 356px; --gap: 14px;"
          tabindex="-1"
        >
          <li
            aria-atomic="true"
            aria-live="polite"
            class="group toast flex justify-between group-[.toaster]:bg-background group-[.toaster]:text-foreground group-[.toaster]:border-border group-[.toaster]:shadow-lg [&>div>div[data-title]]:font-semibold [&>button]:min-w-[87px]"
            data-dismissible="true"
            data-expanded="false"
            data-front="true"
            data-index="0"
            data-mounted="true"
            data-promise="false"
            data-removed="false"
            data-sonner-toast=""
            data-styled="true"
            data-swipe-out="false"
            data-swiping="false"
            data-visible="true"
            data-x-position="right"
            data-y-position="bottom"
            role="status"
            style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
            tabindex="0"
          >
            <div
              class=""
              data-content=""
            >
              <div
                class=""
                data-title=""
              >
                Elo o cartão do brasileiro
              </div>
              <div
                class="group-[.toast]:text-muted-foreground"
                data-description=""
              >
                Elo o cartão do brasileiro
              </div>
            </div>
            <button
              class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none text-base font-medium antialiased leading-4 bg-primary text-primary-foreground shadow hover:bg-green-dark-400 disabled:bg-neutral-300 disabled:text-neutral-500 py-2 px-4 min-w-[152px] min-h-[24px] border-lg"
            >
              Desfazer
            </button>
          </li>
        </ol>
      </section>
      <button>
        Abrir
      </button>
    </div>
  </div>
</body>
`;
